crate_universe/private/module_extensions/cargo_bazel_bootstrap.bzl

```diff
@@ -0,0 +1,55 @@
+"""Module extension for bootstrapping cargo-bazel."""
+
+load("//crate_universe:deps_bootstrap.bzl", _cargo_bazel_bootstrap_repo_rule = "cargo_bazel_bootstrap")
+
+def _cargo_bazel_bootstrap_impl(_):
+    _cargo_bazel_bootstrap_repo_rule(
+        rust_toolchain_cargo_template = "@rust_host_tools//:bin/{tool}",
+        rust_toolchain_rustc_template = "@rust_host_tools//:bin/{tool}",
+    )
+
+cargo_bazel_bootstrap = module_extension(
+    implementation = _cargo_bazel_bootstrap_impl,
+)
+
+def get_cargo_bazel_runner(module_ctx):
+    """A helper function to allow executing cargo_bazel in module extensions.
+
+    Args:
+        module_ctx: The module extension's context.
+
+    Returns:
+        A function that can be called to execute cargo_bazel.
+    """
+
+    # Until you call module_ctx.path, bazel has no way of knowing that it needs
+    # to execute the rust_host_tools and cargo_bazel_bootstrap repository rules.
+    # Thus, if I had a tag lockfile = "@repo//:Cargo.lock", or similar, then
+    # calling this as the first thing in a module extension allows us to execute
+    # the @rust_host_tools repo rule before @repo finishes executing.
+    cargo_path = str(module_ctx.path("@rust_host_tools//:bin/cargo"))
+    rustc_path = str(module_ctx.path("@rust_host_tools//:bin/rustc"))
+    cargo_bazel = module_ctx.path("@cargo_bazel_bootstrap//:cargo-bazel")
+
+    def run(args, env = {}, timeout = 600):
+        final_args = [cargo_bazel]
+        final_args.extend(args)
+        final_args.extend([
+            "--cargo",
+            cargo_path,
+            "--rustc",
+            rustc_path,
+        ])
+        result = module_ctx.execute(
+            final_args,
+            environment = dict(CARGO = cargo_path, RUSTC = rustc_path, **env),
+            timeout = timeout,
+        )
+        if result.return_code != 0:
+            if result.stdout:
+                print("Stdout:", result.stdout)  # buildifier: disable=print
+            pretty_args = " ".join([str(arg) for arg in final_args])
+            fail("%s returned with exit code %d:\n%s" % (pretty_args, result.return_code, result.stderr))
+        return result
+
+    return run
```

Comment:
Is `get_cargo_bazel_runner` used anywhere? If not it should probably be added later so that this doesn't become stale/dead code.

Reply:
It wasn't used anywhere - I've removed it now